Document 6.x SDK javascript

ballinette opened the pull request:
Document changes introduced by https://github.com/kuzzleio/sdk-javascript/pull/243
fix #350 

Can be previewed here: https://writer-ape-62301.netlify.com

ballinette commented on src/assets/stylesheets/side-code/screen.scss:
```diff
@@ -80,6 +80,32 @@
           }
         }
 
+        .noside {
```
hack to enable page /sdk-reference/essentials/notifications/ to display both `<pre>` in right side (sample codes) and within the main content column (JSON examples)

dbengsch commented on src/assets/stylesheets/side-code/screen.scss:
```diff
@@ -80,6 +80,32 @@
+              overflow-x: auto;
+              padding: 25px;
+              color: #383a42;
+              background: #fafafa;            }
```
bad bracket placement

dbengsch commented on src/guide/getting-started/index.md:
```diff
@@ -107,16 +107,18 @@ const Kuzzle = require('kuzzle-sdk')
-    console.error(err.message)
-  })  
-  .finally(() => kuzzle.disconnect())
+kuzzle.addListener('connected', () => {
```
Does that mean that it will try to create the index at each re-connection (if auto-reconnect is enabled) ?

ballinette replied:
the sample script will disconnect and exit immedialty after index creation, so there cannot happen.
But you're right, it can make some confusion... should be replaced by `once`()

dbengsch commented on src/guide/getting-started/index.md:
```diff
@@ -149,15 +151,17 @@ const kuzzle = new Kuzzle('localhost', {defaultIndex: 'playground'})
-    console.error(err.message)
-  })
-  .finally(() => kuzzle.disconnect())
+kuzzle.addListener('connected', () => {
```
(same as above for the document) ?

scottinet commented on src/sdk-reference/collection/room.md:
```diff
@@ -13,10 +13,11 @@ title: room
-  .room()
-  .renew({in: {field: ['some', 'new', 'filter']}}, function (err, res) {
+  .room({in: {field: ['some', 'filter']}})
+  .on('document', functon (err, res) {
```
`function`

scottinet commented on src/sdk-reference/collection/room.md:
```diff
@@ -30,8 +31,8 @@ JSONObject filters = new JSONObject()
-  .room()
-  .renew(filters, new ResponseListener<NotificationResponse>() {
+  .room(filters)
+  .on(new ResponseListener<NotificationResponse>() {
```
missing event name

ballinette commented:
@scottinet I've added a section for `KuzzleEventEmitter` and moved event-related methods from `Kuzzle` to there: https://writer-ape-62301.netlify.com/sdk-reference/event-emitter/
+ added a reference from `Kuzzle` and `Room` contructors to this new section

y-abs commented:
Feel weird to read "Some How To's IOT" or "Some How To's about MANAGING DATA"

Why not just have the title above "Kuzzle How To", with categories "IOT", and "Managing data" ?
